infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/DataSender.java

```diff
@@ -30,6 +33,12 @@
      */
     CompletableFuture<LogReplicationEntry> send(List<LogReplicationEntry> messages);
 
+    /**
+     * Used by Snapshot Full Sync to poll the receiver's status.
+     * @return
+     */
+    public CompletableFuture<LogReplicationQueryMetadataResponse> sendQueryMetadataRequest() throws ExecutionException, InterruptedException;
```

Comment:
We can get rid of the public and the exceptions. 

`CompletableFuture<LogReplicationQueryMetadataResponse> sendQueryMetadataRequest()`

Reply:
+1